application/models/google_cloud_bucket.py

```diff
@@ -0,0 +1,34 @@
+import logging
+
+import structlog
+from google.cloud import storage
+
+log = structlog.wrap_logger(logging.getLogger(__name__))
+
+
+class GoogleCloudSEFTCIBucket:
+    def __init__(self, config):
+        self.project_id = config["GOOGLE_CLOUD_PROJECT"]
+        self.bucket_name = config["SEFT_CI_BUCKET_NAME"]
+        self.client = storage.Client(project=self.project_id)
+        self.bucket = self.client.bucket(self.bucket_name)
+        self.prefix = config["SEFT_CI_BUCKET_FILE_PREFIX"]
```

Comment:
There's one thing missing from this, we need to manage the keys so we can do customer supplied keys with google. If you refer to the design it is detailed on there.

Reply:
As discussed we'll spin this out into a separate card as it's going in feature flagged off